### ISSUE TITLE
Decide on using title vs aria-label

### ISSUE THREAD

**Issue by xi** [liqd/a4-meinberlin#133]
When we have buttons that only contain an icon and maybe in some other cases, we need to provide an alternative text ([WCAG 1.1.1](https://www.w3.org/TR/WCAG20/#text-equiv-all)). There are a lot of options for that, but currently we use of of these two:

- `aria-label`: `<button><i class="fa" aria-label="alt text"></i></button>`
  - Common user agents ignore this attribute
  - In the [WAI ARIA Text Alternative Computation Algorithm](https://www.w3.org/TR/accname-aam-1.1/#h-mapping_additional_nd_te) this attribute has a very high priority.
- `title`: `<button title="alt text"><i class="fa" aria-hidden="true"></i></button>`
  - Many user agents show the label as a tooltip on (long) hover.
  - The [HTML5 spec](https://www.w3.org/TR/html/dom.html#the-title-attribute) warns that "many user agents do not expose the [title] attribute in an accessible manner".
  - In the [WAI ARIA Text Alternative Computation Algorithm](https://www.w3.org/TR/accname-aam-1.1/#h-mapping_additional_nd_te) the title attribute is used, but has a very low priority.

Both options have pros and cons, but we should have clear rules on when to use which one.

**Comment by vellip** [comment]
As said, these two are for different usecases:
* aria-label is something that definitely gets read by screen readers. It's only accessible to screen readers.
* title gets read by some screen readers but not all and throughout the a11y community it's not advised to use title. However it does have the neat feature of a simple tooltip for further explanations.

I think there are several ways how to go on with this:
1. We only use aria-label and hope that icons are explanatory enough for sighted people.
2. We only use title and live with the fact that not every title in every situation always gets read by screen readers.
3. We use aria-label and use title selectively on cases where the context is not enough to always understand an action (i.e. ambiguous icons).
4. We always use aria-label and title together. 
    * We could write a template tag for that which generates a string for both attributes.
    * Titles would always better be placed on the button since the tooltip should be shown for the entire element and not just an item
    * I believe it doesn't matter where we put the aria-label, so we could also place it on the button.

For No. 4 I'd say the appropriate markup would be:
```html
<button {% title "alt text" %}>
    <i class="fa" aria-hidden="true"></i>
</button>
```

Thinking about it, I'd prefer 3 or 4.

**Comment by xi** [comment]
> We use aria-label and use title selectively on cases where the context is not enough to always understand an action (i.e. ambiguous icons).

I think this is the way to go. I found the following cases:

- Landmarks should have `aria-label`. No title is needed.
- On buttons that only contain an icon, the icon should have an `aria-label` and the button should have a `title` with the same content.
- On buttons that contain both text and icon where text and icon means the same thing (e.g. "X Close"), the icon should be `aria-hidden`.

There are some more cases.
